### PR TITLE
Ab tensorize speed up

### PR DESCRIPTION
**Pull request by abaumann**
Modified how tensorize.sh works to speed it up about 10x - originally this was running a batch of samples in serial in a docker container, and running multiple docker containers in parallel.  I noticed this was using very little cpu, memory, or disk.  Changed it to istead use xargs -P to run in parallel in a single docker container - the CPU/disk/mem all stay at nearly maxed out with the right combination of settings (e.g. on a 22 core machine I did 20 in parallel).  Memory usage was low, so I'll update docs about what machine specs to use/what flags to use.  

As part of this, I added a few other things:

- a script to validate the generated tensors - some times something can happen and they are corrupt, that script helps you find the corrupt ones
- changed the pip installs of ml4h into the docker to be --quiet mode so it's less noisy (if it fails you will still see that)
- added hdf5-tools to the docker for use in validating or looking at hd5 files
- changed the tensor writer to open the file in write rather than append mode and throw an error if someone tries to overwrite a tensor that already exists - it had been happily appending to existing files with the same data, which would cause downstream effects

### CODE REVIEW

File: scripts/validate_tensors.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+# BAD - /mnt/disks/tensors/ukb5678.hd5
+
+
+INPUT_TENSORS_DIR=$1
+NUMBER_OF_THREADS=$2
+
+
+find ${INPUT_TENSORS_DIR}/*.hd5 | \
+    xargs -P ${NUMBER_OF_THREADS} -I {} \
+        bash -c "h5dump -n {} | (grep -q 'HDF5 \"{}\"' && echo 'OK - {}' || echo 'BAD - {}')"
```
Comment (lucidtronix) [inline scripts/validate_tensors.sh:15]: How does the script know if an HD5 bad here?

File: scripts/tf.sh [hunk 2]
```diff
@@ -184,6 +184,6 @@ ${GPU_DEVICE} \
 -v ${WORKDIR}/:${WORKDIR}/ \
 -v ${HOME}/:${HOME}/ \
 ${MOUNTS} \
-${DOCKER_IMAGE} /bin/bash -c "pip3 install --upgrade pip
-pip install ${WORKDIR};
+${DOCKER_IMAGE} /bin/bash -c "pip install --quiet --upgrade pip
+pip install --quiet ${WORKDIR};
```
Comment (lucidtronix) [inline scripts/tf.sh:188]: <3

**Review by lucidtronix — APPROVED** [review]
Thanks @abaumann one questions inline but feel free to merge!

File: scripts/validate_tensors.sh (hunk 1 above)
Comment (abaumann) [reply scripts/validate_tensors.sh:15]: yeah this was one thing I wanted to check about, but basically it looks for output which starts with "HDF5"  and then the file path and then information inside the file, so all it really does is check that h5dump doesn't exit - that seems to indicate it's not corrupt, but doesn't check anything about the actual contents.

**Pull request merged**